x-pack/test/security_solution_api_integration/test_suites/entity_analytics/entity_store/trial_license_complete_tier/host_transform.ts

```diff
@@ -168,3 +216,100 @@ function buildHostTransformDocument(name: string, host: EcsHost): IndexRequest {
   };
   return document;
 }
+
+async function createDocumentsAndTriggerTransform(
+  providerContext: FtrProviderContext,
+  docs: HostTransformTestDocuments,
+  docCount: number
+): Promise<void> {
+  const retry = providerContext.getService('retry');
+  const es = providerContext.getService('es');
+
+  const { count, transforms } = await es.transform.getTransformStats({
+    transform_id: HOST_TRANSFORM_ID,
+  });
+  expect(count).to.eql(1);
+  let transform = transforms[0];
+  expect(transform.id).to.eql(HOST_TRANSFORM_ID);
+  const triggerCount: number = transform.stats.trigger_count;
+  const docsProcessed: number = transform.stats.documents_processed;
+
+  for (let i = 0; i < docCount; i++) {
+    const host: EcsHost = {};
+    if (docs.domain?.[i] !== undefined) {
+      host.domain = docs.domain[i];
+    }
+    if (docs.hostname?.[i] !== undefined) {
+      host.hostname = docs.hostname[i];
+    }
+    if (docs.id?.[i] !== undefined) {
+      host.id = docs.id[i];
+    }
+    host.os = {};
+    if (docs.osName?.[i] !== undefined) {
+      host.os.name = docs.osName[i];
+    }
+    if (docs.osType?.[i] !== undefined) {
+      host.os.type = docs.osType[i];
+    }
+    if (docs.mac?.[i] !== undefined) {
+      host.mac = docs.mac[i];
+    }
+    if (docs.arch?.[i] !== undefined) {
+      host.architecture = docs.arch[i];
+    }
+    if (docs.ip?.[i] !== undefined) {
+      host.ip = docs.ip[i];
+    }
+
+    const { result } = await es.index(buildHostTransformDocument(docs.name, host));
+    expect(result).to.eql('created');
+  }
+
+  // Trigger the transform manually
+  const { acknowledged } = await es.transform.scheduleNowTransform({
+    transform_id: HOST_TRANSFORM_ID,
+  });
+  expect(acknowledged).to.be(true);
+
+  await retry.waitForWithTimeout('Transform to run again', TIMEOUT_MS, async () => {
+    const response = await es.transform.getTransformStats({
+      transform_id: HOST_TRANSFORM_ID,
+    });
+    transform = response.transforms[0];
+    expect(transform.stats.trigger_count).to.greaterThan(triggerCount);
+    expect(transform.stats.documents_processed).to.greaterThan(docsProcessed);
+    return true;
+  });
+}
+
+interface HostTransformTestDocuments {
+  name: string; // required
+
+  domain: string[] | undefined;
+  hostname: string[] | undefined;
+  id: string[] | undefined;
+  osName: string[] | undefined;
+  osType: string[] | undefined;
+  mac: string[] | undefined;
+  arch: string[] | undefined;
+  ip: string[] | undefined;
+}
+
+interface HostTransformResult {
+  host: HostTransformResultHost;
+}
```

Comment:
Maybe it's better just to use `EcsHost` (from '@elastic/ecs') instead of creating the `HostTransformResultHost` interface

Reply:
The result of host transform does not match `EcsHost`, since it can have multiple collected values for single-value fields.
`ECS .host.ip: string` to `HostTransformResult .host.ip: string[]`